dspace-server-webapp/src/test/java/org/dspace/app/rest/hdlresolver/HdlResolverRestControllerIT.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.hdlresolver;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.sql.SQLException;
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.builder.CollectionBuilder;
+import org.dspace.builder.CommunityBuilder;
+import org.dspace.builder.ItemBuilder;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.hamcrest.core.StringContains;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Vincenzo Mecca (vins01-4science - vincenzo.mecca@4science.com)
+ *
+ */
+public class HdlResolverRestControllerIT extends AbstractControllerIntegrationTest {
+
+    /**
+     * This method will be run before every test as per @Before. It will initialize
+     * resources required for the tests.
+     *
+     * Other methods can be annotated with @Before here or in subclasses but no
+     * execution order is guaranteed
+     */
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+    }
+
+    /**
+     * Verifies that any null hdlIdentifier returns a
+     * <code>HttpStatus.BAD_REQUEST</code>
```

Comment:
The test does something else than the comment suggest. Can you please fix the comment? The test is checking if a handle is resolved correctly, the comment suggest it would check a null handle and expect a 400 http status code.